Util/src/main/java/io/deephaven/util/SaturationArithmetic.java

```diff
@@ -0,0 +1,25 @@
+package io.deephaven.util;
+
+// See http://locklessinc.com/articles/sat_arithmetic/
```

Comment:
If we're going to reference this site, we should be clear that we aren't using any code from there, just using it to understand the concepts of saturated arithmetic.